pkg/api/norman/customization/cluster/actionhandler.go

```diff
@@ -100,6 +98,21 @@ func (a ActionHandler) ClusterActionHandler(actionName string, action *types.Act
 			return httperror.NewAPIError(httperror.PermissionDenied, "can not rotate certificates")
 		}
 		return a.RotateCertificates(actionName, action, apiContext)
+	case v32.ClusterActionRotateEncryptionKey:
+		if !canUpdateCluster() {
+			return httperror.NewAPIError(httperror.PermissionDenied, "can not rotate encryption key")
+		}
+		enabled, err := a.secretEncryptionEnabled(apiContext.ID)
+		if err != nil {
+			return httperror.NewAPIError(httperror.ClusterUnavailable, err.Error())
+		}
+		if !enabled {
+			return httperror.NewAPIError(httperror.InvalidAction, "encryption is disabled")
+		}
+		if !canBackupEtcd() {
```

Comment:
could we move this permission check after canUpdateCluster? all access stuff would come first imo

Reply:
Good point, I'll make that change